scripts/normalize-by-median.py

```diff
@@ -126,9 +143,40 @@ def handle_error(error, output_name, input_name, fail_save, htable):
         print >> sys.stderr, '** ERROR: problem removing corrupt filtered file'
 
 
+@contextmanager
+def FailSafe(ifile, ofile, save_on_fail, ht, force):
+    global corrupt_files, total, discarded, total_acc, discarded_acc
+    try:
+        yield
```

Comment:
the `@contextmanager` decorator turns a generator into a context manager. When the block executes the interpreter replaces the `yield` with whatever's within the relevant `with` block. In this case from here: https://github.com/dib-lab/khmer/pull/1010/files#diff-6b3b058584d44911b3cf3fe21d9c9af4R197; the lines 199 & 200 are shoved in where the `yield` is and then it's executed as a 'normal' block.

If you meant the context manager it just abstracts out the IOError catching and accounting.